### PR TITLE
fixed data-manager get nearest data point method when keying on strings

### PR DESCRIPTION
**Pull request by clhenrick**
Currently when keying on nominal (string) data types (as for stacked bar) the tooltip will behave wacky as it attempts to use d3-array's bisector to map the mouse x position to the corresponding data value. Rather than use a bisector, it's more straightforward to locate the data point by using `Array.find()`.

### CODE REVIEW

File: src/charts/data-manager.js [hunk 1]
```diff
@@ -195,6 +195,12 @@ export default function DataManager () {
 
   function getNearestDataPoint (_mouseX, _dataObject, _scales, _keyType) {
     const keyFromInvertedX = invertScale(_scales.xScale, _mouseX, _keyType)
+
+    if (_keyType === "string") {
+      // if we are keying on strings, simply find the value via a key match
+      return _dataObject.dataByKey.find(d => d[keys.KEY] === keyFromInvertedX)
```
[the remark this answers is not in the document]
Comment (clhenrick) [reply src/charts/data-manager.js]: @biovisualize are you suggesting that we use a polyfill? Babel only includes one if you include `babel-polyfill` as a dependency

[the remark this answers is not in the document]
Comment (biovisualize) [reply src/charts/data-manager.js]: I don't know. We should test the Immerse PR in IE to see if it's a problem. I just tried to add an Array.find somewhere and compile MapD3, and I see that it's not polyfilled. Maybe adding babel-polyfill would be safer anyway.

[the remark this answers is not in the document]
Comment (clhenrick) [reply src/charts/data-manager.js]: We do in fact include `babel-polyfill` in Immerse (it's in `package.json` and loaded in the webpack config). We don't want to include it here as well as Immerse, so we could advise users of mapd3 to use a polyfill library or add babel-polyfill as a peer dependency.

[the remark this answers is not in the document]
Comment (biovisualize) [reply src/charts/data-manager.js]: Ok right, Immerse will take care of it. It's never clear to me if we have some vendor parts that are compiled separately from Immerse process.

**Pull request merged**